src/Shape/Box.js

```diff
@@ -0,0 +1,355 @@
+import {max, min, quantile} from "d3-array";
+import {nest} from "d3-collection";
+import {select} from "d3-selection";
+
+import {accessor, assign, BaseClass, configPrep, constant, merge, elem} from "d3plus-common";
+
+import Circle from "./Circle";
+import Rect from "./Rect";
+import Whisker from "./Whisker";
+
+const shapes = {Circle, Rect};
+
+/**
+    @class Box
+    @extends BaseClass
+    @desc Creates SVG box based on an array of data.
+*/
+export default class Box extends BaseClass {
+
+  /**
+      @memberof Box
+      @desc Invoked when creating a new class instance, and overrides any default parameters inherited from BaseClass.
+      @private
+  */
+  constructor() {
+
+    super();
+
+    this._medianConfig = {
+      fill: constant("black")
+    };
+    this._orient = accessor("orient", "vertical");
+    this._outlier = accessor("outlier", "Circle");
+    this._outlierConfig = {
+      Circle: {
+        r: accessor("r", 5)
+      },
+      Rect: {
+        height: (d, i) => this._orient(d, i) === "vertical" ? 5 : 20,
+        width: (d, i) => this._orient(d, i) === "vertical" ? 20 : 5
+      }
+    };
+    this._rectConfig = {
+      fill: constant("white"),
+      stroke: constant("black"),
+      strokeWidth: constant(1)
+    };
+    this._rectWidth = constant(50);
+    this._whiskerConfig = {};
+    this._whiskerMode = ["tukey", "tukey"];
+    this._x = accessor("x", 250);
+    this._y = accessor("y", 250);
+
+  }
+
+  /**
+      @memberof Box
+      @desc Draws the Box.
+      @param {Function} [*callback*]
+      @chainable
+  */
+  render() {
+
+    if (this._select === void 0) {
+      this.select(select("body").append("svg")
+        .style("width", `${window.innerWidth}px`)
+        .style("height", `${window.innerHeight}px`)
+        .style("display", "block").node());
+    }
+
+    const outlierData = [];
+
+    const filteredData = nest()
+      .key((d, i) => this._orient(d, i) === "vertical" ? this._x(d, i) : this._y(d, i))
+      .entries(this._data)
+      .map((d, i) => {
+        d.data = merge(d.values);
+        d.orient = this._orient(d.data, d.i);
```

Comment:
The linting error made me notice this line, where you use `d.i` before declaring it on line 82. Move line 82 above this line.